server/raft.go

```diff
@@ -2268,6 +2275,15 @@ func (n *raft) storeToWAL(ae *appendEntry) error {
 		fmt.Printf("[%s] n is %+v\n\n", n.s, n)
 		fmt.Printf("[%s] n.catchup is %+v\n", n.s, n.catchup)
 		fmt.Printf("[%s] n.wal is %+v\n", n.s, n.wal.State())
+		if state := n.wal.State(); state.Msgs > 0 {
+			for index := state.FirstSeq; index <= state.LastSeq; index++ {
+				nae, _ := n.loadEntry(index)
+				fmt.Printf("INDEX %d is %+v\n", index, nae)
```

Comment:
some debug left here

Reply:
Yes on purpose to print out when the condition trips. There are several nuanced bugs I believe that can lead to this condition.